src/test/java/uk/gov/pay/ledger/transaction/service/TransactionServiceTest.java

```diff
@@ -159,6 +159,22 @@ public void shouldListTransactionsWithCorrectQueryParamsForPaginationLinks() {
         assertCorrectPaginationQueryParams(transactionSearchResponse);
     }
 
+    @Test
+    public void searchTransactionsWithoutParent_shouldThrowNotFoundException_forInvalidPaginationParams() {
+        List<TransactionEntity> transactionViewList = TransactionFixture.aTransactionList(gatewayAccountId, 10);
+        when(mockTransactionDao.searchTransactions(any(TransactionSearchParams.class))).thenReturn(transactionViewList);
+        when(mockTransactionDao.getTotalForSearch(any(TransactionSearchParams.class))).thenReturn(10L);
+        thrown.expect(WebApplicationException.class);
+        thrown.expectMessage("the requested page not found");
+
+        searchParams.setPageNumber(100L);
```

Comment:
Wouldn't `2` be enough to cause the failure? (we have 10 transactions and  MAX_DISPLAY_SIZE of 500);
Ideally we should set the displaySize to 10 (or something) to test the edge case.

Reply:
👍 Updated